andretkachenko/nexus-bot
# Expand documentation

# Description

Add more extensive help documentation:
- split previous help embed into 2
- add new help embed for each existing command
- add new help embeds for the most frequently asked questions

Fixes #116 

## Type of change


# How Has This Been Tested?

1. Create voice channel `1`
2. Join `1`
3. Delete `1`
4. Ensure `1-text` was deleted

1. Create voice channel `1`
2. Create voice channel `2`
3. Join `1`
4. Join `2`
5. Map `1` and `2-text` via `map` command with override === 0
6. Observe `Voice Channel already has mapped Text Channel` error at the text channel where commands were sent

1. Create voice channel `1`
2. Create voice channel `2`
3. Join `1`
5. Map `2` and `1-text` via `map` command with override === 0
6. Observe `Text Channel is already mapped to a Voice Channel` error at the text channel where commands were sent

1. Create voice channel `1`
2. Create voice channel `2`
3. Join `1`
4. Join `2`
5. Map `1` and `2-text` via `map` command with override === 1
6. Observe `The Voice Channel already had a mapped Text Channel. Changed to map with the Text Channel #{id}` notification at the text channel where commands were sent
7. Observe `The Text Channel was already mapped to a Voice Channel. Deleted mapping with the Voice Channel #{id}` notification at the text channel where commands were sent
8. Observe `Mapped the Text Channel with the Voice Channel #{id}` notification at the text channel where commands were sent

1. type invalid command input: `sudo map 0 qwe qwe`
2. Observe `Error processing command - specified channel ID does not belong to a voice channel` error
3. Observe `invalid Voice Channel ID` error

1. Create voice channel `1`
2. Try to map `1` with invalid Text Channel ID
3. Observe `invalid Text Channel ID` error

1. Type `sudo ping`
2. Ensure bot responded with `alive and waiting for your commands`

1. Type `sudo help`
2. Ensure bot responded with valid help embed

1. Type `sudo help commands`
2. Ensure bot responded with valid help embed

1. Type `sudo help channelid`
2. Ensure bot responded with valid help embed

1. Type `sudo help ignore`
2. Ensure bot responded with valid help embed

1. Type `sudo help map`
2. Ensure bot responded with valid help embed

1. Type `sudo help ping`
2. Ensure bot responded with valid help embed

1. Type `sudo help preserve`
2. Ensure bot responded with valid help embed

1. Type `sudo help skip`
2. Ensure bot responded with valid help embed

1. Type `sudo help whoops`
2. Ensure bot responded with valid help embed

1. Type `sudo help write`
2. Ensure bot responded with valid help embed

1. Type `sudo write test` into a text channel
2. Ensure `test` message was sent by the bot

1. Add voice channel `channel1` to the ignore list
2. Ensure new record was added at DB
3. Ensure new DB record is valid
4. Join `channel1` voice channel
5. Ensure `channel1-text` wasn't created at the `Nexus channels` category

1. Remove voice channel `channel1` from the ignore list
2. Ensure valid DB record was deleted
4. Join `channel1` voice channel
5. Ensure `channel1-text` wasn't created at the `Nexus channels` category

1. Add user `testuser` to the skip list
2. Ensure new record was added at DB for `testuser`
3. Ensure new DB record is valid
4. Join `channel1` voice channel with `testuser`
5. Ensure `channel1-test` visibility settings for the `testuser` didn't change

1. Remove user `testuser` from the skip list
2. Ensure valid DB record was deleted
3. Join `channel1` voice channel with `testuser`
4. Ensure `channel1-test` visibility settings for the `testuser` has changed

1. Add role `nexusTestSkip` to the skip list
3. Ensure a new record was added at DB for `nexusTestSkip`
4. Ensure new DB record is valid
5. Add role `nexusTestSkip` for the `testuser`
5. Join `channel1` voice channel with `testuser`
6. Ensure `channel1-test` visibility settings for the `testuser` didn't change

1. Remove role `nexusTestSkip` from the skip list
2. Ensure valid DB record was deleted
3. Join `channel1` voice channel with `testuser`
4. Ensure `channel1-test` visibility settings for the `testuser` has changed

1. Add user `testuser` and role `nexusTestSkip` to the skip list
3. Ensure a new record was added at DB for `nexusTestSkip`
4. Ensure new DB record is valid
5. Join `channel1` voice channel with `testuser`
6. Ensure `channel1-test` visibility settings for the `testuser` didn't change

1. Create voiceChannel1 and voiceChannel2
2. Join both channels to create new linked text channels
3. Ensure new records were added at DB
4. Ensure new DB records are valid
5. Set preserve option for voiceChannel1
6. Ensure required DB record was updated correctly
7. Join voiceChannel1
8. Write test message in textChannel1
9. Join voiceChannel2
10. Ensure textChannel1 wasn't cleared

1. Create voiceChannel1 and voiceChannel2
2. Join both channels to create new linked text channels
3. Join voiceChannel2
4. Write test message in textChannel2
5. Join voiceChannel1
6. Ensure textChannel2 was cleared

**Test Configuration**:
* Firmware version: 0.0.307
* Hardware: Win 10
* Toolchain: VSCode, Discord, Google Chrome

# Checklist:
